### ISSUE TITLE
Categories for ingredients

### ISSUE THREAD

**Issue by rettenbs** [nazgul26/PHPRecipebook#71]
Hi,

it would be nice if it is possible to sort ingredients of a recipe into categories. For example for a cake, you have the cake base and the cream (and maybe something for decorations).

I think the easiest implementation would be to have some special "ingredient" that acts as a separator.

Best,
Sebastian

**Comment by rettenbs** [comment]
Hi,

it would be nice if it is possible to sort ingredients of a recipe into categories. For example for a cake, you have the cake base and the cream (and maybe something for decorations).

I think the easiest implementation would be to have some special "ingredient" that acts as a separator.

Best,
Sebastian

**Comment by ilveroluca** [comment]
The tool supports "related recipes", which I believe are supposed to address this use case.  For a cake, you could enter a related recipe for the cream.

**Comment by nazgul26** [comment]
Yes.  The related recipes are intended to be used for shared 'recipes' like frosting, broth, etc.  Thanks.

**Comment by rettenbs** [comment]
The related recipes are nice but for some of my recipes it would be quite an overhead to create 2 or 3 recipes to put the ingredients into categories.

I started working on categories. At the moment I have one (hidden) category for each recipe:

https://github.com/rettenbs/PHPRecipebook/commit/3bed96da72763355af0da3f1da04f120a366552f
https://github.com/rettenbs/PHPRecipebook/commit/13d923180a3bb17722bec08998bfa89a7be54a5b

View, editing and searching works, as well as the shopping lists. Not sure if their is another place where the mapping between ingredients and recipes is used.

To test the patches, one has to upgrade the database schema. I updated the Config/schema.php but I am not sure how to (semi-)automatically execute the 3 SQL statements one needs to migrate to the new schema.

At the moment, all changes are hidden. Next thing on my todo list is to modify the recipe edit form such that the user can add multiple groups.
